### PR TITLE
Fix NVTX functions compatibility with torch._dynamo

### PR DESCRIPTION
**Pull request by zsnoob**
## Problem Solved

This PR resolves the incompatibility between NVTX functions and torch._dynamo. When attempting to use NVTX profiling tools within code compiled with torch.compile(), the following error occurs:

```
torch._dynamo.exc.Unsupported: torch.* op returned non-Tensor int call_function <function range_push at 0x...>
```

The root cause is that torch._dynamo requires all function calls within a compiled graph to return tensor types, but NVTX functions return integers, objects, or None.

## Changes

- Added a global toggle system to enable/disable tensor returns for NVTX functions
- Implemented a decorator to handle type conversion automatically
- Enhanced all NVTX functions to support tensor return mode
- Added clear documentation and type annotations
- Maintained backward compatibility with existing code

## Impact on Existing Functionality

This change has **zero impact** on existing functionality when used normally. The default behavior remains unchanged, and all functions continue to return their original types.

Only when explicitly enabled via `torch.utils.nvtx.enable_tensor_returns()` will the functions return tensor types instead. This opt-in approach ensures no disruption to existing code.

## Testing

- Added comprehensive unit tests that verify:
  - Default behavior is preserved
  - Tensor return mode correctly converts all return types to tensors
  - Switching between modes works as expected

## Usage Example
```python
# Enable tensor returns for dynamo compatibility
torch.cuda.nvtx.enable_tensor_returns()

# Use NVTX functions in dynamo-compiled code
# All functions now return tensors

# with torch.compile context
with torch.cuda.nvtx.range("my_range"):
    pass

# Disable tensor returns to restore original behavior
torch.cuda.nvtx.disable_tensor_returns()
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Comment by zsnoob** [comment]
> release notes:

release notes: dynamo

**Comment by zsnoob** [comment]
> ## 🔗 Helpful Links
> ### 🧪 See artifacts and rendered test results at [hud.pytorch.org/pr/149555](https://hud.pytorch.org/pr/149555)
> * 📄 Preview [Python docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/149555/index.html)
> * 📄 Preview [C++ docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/149555/cppdocs/index.html)
> * ❓ Need help or want to give feedback on the CI? Visit the [bot commands wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands) or our [office hours](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)
> 
> Note: Links to docs will display an error until the docs builds have been completed.
> 
> This comment was automatically generated by Dr. CI and updates every 15 minutes.

release notes: cuda

**Comment by zsnoob** [comment]
> ## 🔗 Helpful Links
> ### 🧪 See artifacts and rendered test results at [hud.pytorch.org/pr/149555](https://hud.pytorch.org/pr/149555)
> * 📄 Preview [Python docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/149555/index.html)
> * 📄 Preview [C++ docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/149555/cppdocs/index.html)
> * ❓ Need help or want to give feedback on the CI? Visit the [bot commands wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands) or our [office hours](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)
> 
> Note: Links to docs will display an error until the docs builds have been completed.
> 
> This comment was automatically generated by Dr. CI and updates every 15 minutes.

release notes: dynamo

**Comment by zsnoob** [comment]
> ## 🔗 Helpful Links
> ### 🧪 See artifacts and rendered test results at [hud.pytorch.org/pr/149555](https://hud.pytorch.org/pr/149555)
> * 📄 Preview [Python docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/149555/index.html)
> * 📄 Preview [C++ docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/149555/cppdocs/index.html)
> * ❓ Need help or want to give feedback on the CI? Visit the [bot commands wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands) or our [office hours](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)
> 
> Note: Links to docs will display an error until the docs builds have been completed.
> 
> This comment was automatically generated by Dr. CI and updates every 15 minutes.

release notes: cuda

**Comment by zsnoob** [comment]
> @pytorchbot label "release notes: cuda"

@pytorchbot label "release notes: cuda"

**Comment by zsnoob** [comment]
Yeah, I think i need resolve some test and grammar issues first
